sdks/python/apache_beam/metrics/metric_test.py

```diff
@@ -127,6 +134,35 @@ def test_distribution_empty_namespace(self):
     with self.assertRaises(ValueError):
       Metrics.distribution("", "names")
 
+  @attr('ValidatesRunner')
+  def test_user_counter_using_pardo(self):
+    class SomeDoFn(beam.DoFn):
+      """A custom dummy DoFn using yield."""
+      def __init__(self):
+        self.user_counter_elements = metrics.Metrics.counter(
+          self.__class__, 'metrics_user_counter_element')
+
+      def process(self, element):
+        self.user_counter_elements.inc()
+        yield element
+
+    pipeline = TestPipeline()
+    nums = pipeline | 'Input' >> beam.Create([1, 2, 3, 4])
+    results = nums | 'ApplyPardo' >> beam.ParDo(SomeDoFn())
+
+    res = pipeline.run()
+    res.wait_until_finish()
+    metric_results = (
+      res.metrics().query(MetricsFilter()
+        .with_name('metrics_user_counter_element')))
+    outputs_counter = metric_results['counters'][0]
+    assert_that(results, equal_to([1, 2, 3, 4]))
```

Comment:
This must be applied before the pipeline is run.

Reply:
doen.